perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_elixirBonaFideStatus.java

```diff
@@ -0,0 +1,145 @@
+				String oldAttrValue = (String) oldAttr.getValue();
+				if (!oldAttrValue.isEmpty()) {
+					return oldAttr;
+				}
+			}
+		} catch (WrongAttributeAssignmentException e) {
+			log.error("Cannot read {} from user {}", getAttributeDefinition().getFriendlyName(), user, e);
+		} catch (AttributeNotExistsException e) {
+			//do nothing - if this attribute does not exist it is OK, we will create it
+		}
+
+		//try to get value from 'elixirBonaFideStatusREMS': if not empty, we have bona_fide
+		try {
+			Attribute statusAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, A_U_D_userBonaFideStatusRems);
+			if (statusAttr != null && statusAttr.getValue() != null) {
+				String statusAttrValue = (String) statusAttr.getValue();
+				if (!statusAttrValue.isEmpty()) {
+					attribute.setValue(URL);
+					return attribute;
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read {} from user {}", USER_BONA_FIDE_STATUS_REMS_ATTR_NAME, user, e);
+		}
+
+		//try to get value from 'eduPersonScopedAffiliations': if has faculty@..., we have bona_fide
+		try {
+			Attribute affiliationsAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, A_U_V_userAffiliations);
+			if (affiliationsAttr != null && affiliationsAttr.getValue() != null) {
+				ArrayList<String> affiliationsValue = affiliationsAttr.valueAsList();
+				for (String val : affiliationsValue) {
+					if (val.startsWith("faculty@")) {
+						attribute.setValue(URL);
+						return attribute;
+					}
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read {} from user {}", USER_AFFILIATIONS_ATTR_NAME, user, e);
+		}
+
+		//try to get value from publications: if has thanks to ELIXIR, we have bona_fide
+		try {
+			Attribute publicationsAttr = sess.getPerunBl().getAttributesManagerBl().getAttribute(sess, user, A_U_D_userPublications);
+			if (publicationsAttr != null && publicationsAttr.getValue() != null) {
+				Map<String, String> publications = publicationsAttr.valueAsMap();
+				if (publications.containsKey(ELIXIR_KEY)) {
+					String value = publications.get(ELIXIR_KEY);
+					try {
+						int count = Integer.parseInt(value);
+						if (count > 0) {
+							attribute.setValue(URL);
+						}
+					} catch (NumberFormatException ex) {
+						log.error("Attribute " + A_U_D_userPublications +" has wrong value for key " + ELIXIR_KEY, ex);
+					}
+				}
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Cannot read {} from user {}", USER_PUBLICATIONS_ATTR_NAME, user, e);
+		}
+
+		return attribute;
+	}
+
+	@Override
+	public List<String> getDependencies() {
+		List<String> dependencies = new ArrayList<>();
+		dependencies.add(A_U_D_userBonaFideStatusRems);
+		dependencies.add(A_U_V_userAffiliations);
+		dependencies.add(A_U_D_userPublications);
+		return dependencies;
+	}
```

Comment:
There are no dependencies, because dependencies are strictly bound to getAttributeValue or checkAttributeValue methods.